### ISSUE TITLE
commands: Make cleanDestinationDir work even when staticDir doesn't exist

### ISSUE THREAD

**Issue by Kzer-Za** [gohugoio/hugo#8433]
`hugo v0.81.0+extended linux/amd64 BuildDate=unknown`

Specifically, it seems to be ignored if you have never used this option from CLI. After installing Hugo I just put `cleanDestinationDir: true` in config.yaml of the site I'm trying to make. However, the command `hugo` never deleted any directories or files that were becoming obsolete. For example, I first built the site with the default taxonomies, but later decided that I didn't need them, so I disabled them with

```
disableKinds:
- taxonomy
- term
```

However, after running `hugo` the directories "tags" and "categories" were still in "public". The same was the case with obsolete files/directories in "content". Only after I have commented the option `cleanDestinationDir: true` in the config.yaml and ran `hugo --cleanDestinationDir` from the command line did Hugo build the "public" directory without the obsolete files/directories.

After that, to test the issue, I uncommented the line in the config and ran just `hugo`. But now the option in config is working and the obsolete files are deleted from "public" as they should. It's like something has been initialized by running `hugo --cleanDestinationDir` once.

**Comment by matt9189** [comment]
I am unable to reproduce this error. When I create a new site and run hugo with the default config, it creates the two folders. When I modify the config file with the following:
`
disableKinds = ["taxonomy", "term"]
`
`
cleanDestinationDir = true
`
It removes the two folders from the public folder as expected.

**Comment by jmooring** [comment]
I can reproduce the problem.

TLDR `--cleanDestinationDir` fails to remove items from the `public` directory unless the `static` directory is present.

```
hugo new site mysite
cd mysite
hugo
find . -type d -name categories -o -name tags
```

The `categories` and `tags` directories are present (PASS).

```
hugo --disableKinds "taxonomy,term" --cleanDestinationDir
find . -type d -name categories -o -name tags
```

The `categories` and `tags` directories are gone (PASS).

```
hugo
find . -type d -name categories -o -name tags
```

The `categories` and `tags` directories are present (PASS).

```
rmdir static
hugo --disableKinds "taxonomy,term" --cleanDestinationDir
find . -type d -name categories -o -name tags
```

The `categories` and `tags` directories are present (FAIL).

```
mkdir static
hugo --disableKinds "taxonomy,term" --cleanDestinationDir
find . -type d -name categories -o -name tags
```

The `categories` and `tags` directories are gone (PASS).

**Comment by bep** [comment]
So, this get skipped if there are no static directories.

I suggest that we in that case just do a "walk and Remove" in the `DestionationFs` -- but we need to respect the filter that we apply in the normal case:

https://github.com/gohugoio/hugo/blob/3d5dbdcb1a11b059fc2f93ed6fadb9009bf72673/commands/hugo.go#L653

**Comment by ResamVi** [comment]
With the help of @jmooring's string of commands and with respect to what @bep mentioned I made the pull request that atleast fixes observed failure

**Comment by jmooring** [comment]
I removed the "stale" label and self-assigned this issue.

As I clean up our `rsync` and `rclone` deployment docs to handle cases where users may or may not have a `static` directory, I need to decide between:

* Writing "clear the `public` directory" (without specifying how), or
* Using `rm -rf public` (which I'd really prefer to avoid).

Existing docs (e.g., `getting-started/usage.md`) already use phrases like "clear the contents of the `public` directory" or "clear the `public` directory" without explaining how to do it.

**Comment by jmooring** [comment]
Along the way I discovered that this integration test...

<details>
<summary><code>TestCleanDestinationDir</code></summary>

```go
func TestCleanDestinationDir(t *testing.T) {
	t.Parallel()

	files := `
-- hugo.toml --
disableKinds = ['page','rss','section','sitemap','taxonomy','term']
cleanDestinationDir = CLEAN_DESTINATION_DIR
-- content/_index.md --
---
title: home
---
-- layouts/home.html --
{{ .Title }}
-- public/public.txt --
public.txt
-- static/static.txt --
static.txt
`

	f := strings.ReplaceAll(files, "CLEAN_DESTINATION_DIR", "false")
	b := hugolib.Test(t, f, hugolib.TestOptOsFs())
	b.AssertFileExists("public/index.html", true)
	b.AssertFileExists("public/public.txt", true)
	b.AssertFileExists("public/static.txt", true)

	f = strings.ReplaceAll(files, "CLEAN_DESTINATION_DIR", "true")
	b = hugolib.Test(t, f, hugolib.TestOptOsFs())
	b.AssertFileExists("public/index.html", true)
	b.AssertFileExists("public/public.txt", false)
	b.AssertFileExists("public/static.txt", true)
}
```
</details>

...fails because `hugolib.Build()` doesn't copy the `static` directory or clean the `public` directory. That's only implemented in the CLI's `commands` package, so `hugolib.Test` can't exercise it. 

For now we'll use a testscript, but the above was a head-scratcher for a while.
